File: prisma/seed.ts
import { PrismaClient, MaintenanceStatus, AlertSeverity, AlertType, ActionStatus } from '@prisma/client';

const prisma = new PrismaClient();

async function main() {
    // Vehicles
    const v1 = await prisma.vehicle.upsert({
        where: { licensePlate: 'DXB-12345' },
        update: {},
        create: {
            id: 'v1',
            make: 'Toyota',
            model: 'Hilux',
            type: 'Pickup Truck',
            year: 2022,
            licensePlate: 'DXB-12345',
            vin: 'JTE1234567890',
            currentMileage: 45000,
            status: 'Active',
            registrationExpiry: new Date('2025-12-01'),
            insuranceExpiry: new Date('2025-12-01'),
            registrationLastRenewed: new Date('2024-12-01'),
            insuranceLastRenewed: new Date('2024-12-01'),
        },
    });

    const v2 = await prisma.vehicle.upsert({
        where: { licensePlate: 'DXB-67890' },
        update: {},
        create: {
            id: 'v2',
            make: 'Nissan',
            model: 'Urvan',
            type: 'Van',
            year: 2021,
            licensePlate: 'DXB-67890',
            vin: 'JN11234567890',
            currentMileage: 82000,
            status: 'In Service',
            registrationExpiry: new Date('2024-06-15'),
            insuranceExpiry: new Date('2024-06-15'),
            registrationLastRenewed: new Date('2023-06-15'),
            insuranceLastRenewed: new Date('2023-06-15'),
        },
    });

    const v3 = await prisma.vehicle.upsert({
        where: { licensePlate: 'DXB-54321' },
        update: {},
        create: {
            id: 'v3',
            make: 'Ford',
            model: 'Transit',
            type: 'Van',
            year: 2023,
            licensePlate: 'DXB-54321',
            vin: 'WF01234567890',
            currentMileage: 12000,
            status: 'Active',
            registrationExpiry: new Date('2026-01-20'),
            insuranceExpiry: new Date('2026-01-20'),
            registrationLastRenewed: new Date('2025-01-20'),
            insuranceLastRenewed: new Date('2025-01-20'),
        },
    });

    // Drivers
    const d1 = await prisma.driver.upsert({
        where: { licenseNumber: 'UAE-1234567' },
        update: {},
        create: {
            id: 'd1',
            name: 'Ahmed Al-Farsi',
            licenseNumber: 'UAE-1234567',
            licenseExpiry: new Date('2026-05-10'),
            assignedVehicleId: 'v1',
            contactNumber: '+971501234567',
            licenseLastRenewed: new Date('2021-05-10'),
        },
    });

    const d2 = await prisma.driver.upsert({
        where: { licenseNumber: 'UAE-7654321' },
        update: {},
        create: {
            id: 'd2',
            name: 'John Smith',
            licenseNumber: 'UAE-7654321',
            licenseExpiry: new Date('2024-08-22'),
            assignedVehicleId: 'v2',
            contactNumber: '+971559876543',
            licenseLastRenewed: new Date('2019-08-22'),
        },
    });

    // Garages
    const g1 = await prisma.garage.create({
        data: {
            id: 'g1',
            name: 'AutoPro Service Center',
            location: 'Al Quoz, Dubai',
            contactPerson: 'Mohammed Ali',
            designation: 'Service Manager',
            email: 'mohammed.ali@autopro.ae',
            contactNumber: '+97141234567',
            specialties: ['General Service', 'Tires', 'AC', 'Oil Change'],
            isInternal: false,
        }
    });

    const g2 = await prisma.garage.create({
        data: {
            id: 'g2',
            name: 'Dynatrade',
            location: 'Nadd Al Hamar, Dubai',
            contactPerson: 'Suresh Kumar',
            designation: 'Operations Head',
            email: 'suresh@dynatrade.ae',
            contactNumber: '+97149876543',
            specialties: ['Preventive', 'Corrective'],
            isInternal: false,
        }
    });

    // Maintenance Requests
    await prisma.maintenanceRequest.create({
        data: {
            id: 'MR#241001',
            readableId: 'MR#241001',
            vehicleId: 'v2',
            driverId: 'd2',
            requestDate: new Date('2024-05-20T09:00:00Z'),
            description: 'Engine making strange rattling noise when accelerating.',
            status: MaintenanceStatus.UNDER_MAINTENANCE,
            garageId: 'g2',
            estimatedCost: 1500,
            history: {
                create: [
                    { status: MaintenanceStatus.REQUESTED, date: new Date('2024-05-20T09:00:00Z'), note: 'Request created', actor: 'John Doe (Driver)' },
                    { status: MaintenanceStatus.ACCEPTED, date: new Date('2024-05-20T10:30:00Z'), note: 'Request Accepted', actor: 'Sarah Connor (Fleet Manager)' },
                ]
            },
            comments: {
                create: [
                    { author: 'John Smith', text: 'Noise started this morning.' },
                ]
            }
        }
    });

    await prisma.maintenanceRequest.create({
        data: {
            id: 'MR#241002',
            readableId: 'MR#241002',
            vehicleId: 'v1',
            driverId: 'd1',
            requestDate: new Date('2024-05-24T14:30:00Z'),
            description: 'Periodic maintenance due (50k service).',
            status: MaintenanceStatus.UNDER_ESTIMATION,
        }
    });

    console.log('Seeding completed.');
}

main()
    .then(async () => {
        await prisma.$disconnect();
    })
    .catch(async (e) => {
        console.error(e);
        await prisma.$disconnect();
        process.exit(1);
    });

File: backend/seed/seed.go
package seed

import (
	"fleet360-backend/database"
	"fleet360-backend/logging"
	"fleet360-backend/models"
	"time"

	"go.uber.org/zap"
)

func Seed() {
	log := logging.L()
	// Check if data exists
	var count int64
	database.DB.Model(&models.Vehicle{}).Count(&count)
	if count > 0 {
		log.Info("database already seeded — skipping", zap.Int64("vehicle_count", count))
		return
	}

	log.Info("seeding database")

	// Vehicles
	vehicles := []models.Vehicle{
		{
			Model:              models.Model{ID: "v1"},
			Make:               "Toyota",
			VehicleModel:       "Hilux",
			Type:               "Pickup Truck",
			Year:               2022,
			LicensePlate:       "DXB-12345",
			VIN:                "JTE1234567890",
			CurrentMileage:     45000,
			Status:             "Active",
			RegistrationExpiry: time.Date(2025, 12, 1, 0, 0, 0, 0, time.UTC),
			InsuranceExpiry:    time.Date(2025, 12, 1, 0, 0, 0, 0, time.UTC),
		},
		{
			Model:              models.Model{ID: "v2"},
			Make:               "Nissan",
			VehicleModel:       "Urvan",
			Type:               "Van",
			Year:               2021,
			LicensePlate:       "DXB-67890",
			VIN:                "JN11234567890",
			CurrentMileage:     82000,
			Status:             "In Service",
			RegistrationExpiry: time.Date(2024, 6, 15, 0, 0, 0, 0, time.UTC),
			InsuranceExpiry:    time.Date(2024, 6, 15, 0, 0, 0, 0, time.UTC),
		},
	}
	database.DB.Create(&vehicles)

	// Drivers
	v1 := "v1"
	v2 := "v2"
	drivers := []models.Driver{
		{
			Model:             models.Model{ID: "d1"},
			Name:              "Ahmed Al-Farsi",
			LicenseNumber:     "UAE-1234567",
			LicenseExpiry:     time.Date(2026, 5, 10, 0, 0, 0, 0, time.UTC),
			AssignedVehicleID: &v1,
			ContactNumber:     "+971501234567",
		},
		{
			Model:             models.Model{ID: "d2"},
			Name:              "John Smith",
			LicenseNumber:     "UAE-7654321",
			LicenseExpiry:     time.Date(2024, 8, 22, 0, 0, 0, 0, time.UTC),
			AssignedVehicleID: &v2,
			ContactNumber:     "+971559876543",
		},
	}
	database.DB.Create(&drivers)

	// Garages
	garages := []models.Garage{
		{
			Model:         models.Model{ID: "g1"},
			Name:          "AutoPro Service Center",
			Location:      "Al Quoz, Dubai",
			ContactPerson: "Mohammed Ali",
			Designation:   "Service Manager",
			Email:         "mohammed.ali@autopro.ae",
			ContactNumber: "+97141234567",
			IsInternal:    false,
		},
	}
	database.DB.Create(&garages)

	// Service Requests
	serviceRequests := []models.ServiceRequest{
		{
			Model:       models.Model{ID: "SR-1001"},
			RequestorID: "d1",
			ServiceType: "Vehicle Maintenance Service",
			VehicleID:   &v1,
			Priority:    "High",
			Description: "Brake pads need replacement immediately.",
			Date:        time.Date(2025, 11, 24, 0, 0, 0, 0, time.UTC),
			Status:      "Pending",
			History: []models.History{
				{
					Status: models.StatusRequested,
					Date:   time.Date(2025, 11, 24, 0, 0, 0, 0, time.UTC),
					Note:   "Request created",
					Actor:  "System",
				},
			},
		},
		{
			Model:       models.Model{ID: "SR-1004"},
			RequestorID: "u1",
			ServiceType: "Towing & Recovery Service",
			VehicleID:   &v1,
			Priority:    "High",
			Description: "Vehicle broke down on highway",
			Date:        time.Date(2023, 10, 20, 0, 0, 0, 0, time.UTC),
			Status:      "Resolved",
			History: []models.History{
				{
					Status: models.StatusRequested,
					Date:   time.Date(2023, 10, 20, 0, 0, 0, 0, time.UTC),
					Note:   "Request created",
					Actor:  "System",
				},
				{
					Status: models.StatusCompleted,
					Date:   time.Date(2023, 10, 20, 2, 0, 0, 0, time.UTC),
					Note:   "Towing completed",
					Actor:  "Driver",
				},
			},
		},
	}
	database.DB.Create(&serviceRequests)

	log.Info("database seeded successfully")
}
